### ISSUE TITLE
Tests fail running out of handles

### ISSUE THREAD

**Issue by RussKie** [dotnet/winforms#3150]
**Problem description:**

Having moved to a different pool (#3105), and (temporarily) disabled tests that relied on `RemoteExecutor` (#3116), we are now observing new build failures with the following:
```
    System.Windows.Forms.Tests.ProfessionalColorTableTests.ProfessionalColorTable_Properties_Get_ReturnsExpected(factory: Func`2 { Method = System.Drawing.Color <Properties_TestData>b__1_49(System.Windows.Forms.ProfessionalColorTable), Target = <>c { } }) [FAIL]
      System.Runtime.InteropServices.COMException : The current process has used all of its system allowance of handles for Window Manager objects. (0x80070486)
      Stack Trace:
           at System.Threading.Monitor.ReliableEnter(Object obj, Boolean& lockTaken)
           at System.Threading.Monitor.Enter(Object obj, Boolean& lockTaken)
        /_/src/System.Windows.Forms/src/System/Windows/Forms/NativeWindow.cs(423,0): at System.Windows.Forms.NativeWindow.CreateHandle(CreateParams cp)
        .....
        --- End of stack trace from previous location ---

    System.Windows.Forms.Tests.ScrollableControlTests.ScrollableControl_SetAutoScrollMargin_InvokeWithHandle_Success(autoScroll: False, width: 1, height: 2, expectedAutoScrollMargin: {Width=1, Height=2}, expectedLayoutCallCount: 0) [FAIL]
      System.ComponentModel.Win32Exception : Error creating window handle.
      Stack Trace:
        /_/src/System.Windows.Forms/src/System/Windows/Forms/NativeWindow.cs(491,0): at System.Windows.Forms.NativeWindow.CreateHandle(CreateParams cp)
        .....

    System.Windows.Forms.Tests.MonthCalendarTests.MonthCalendar_SelectionRange_SetWithHandle_GetReturnsExpected(value: SelectionRange: Start: 4/28/2020 12:00:00 AM, End: 4/28/2020 12:00:00 AM, expectedSelectionStart: 2020-04-28T00:00:00.0000000+00:00, expectedSelectionEnd: 2020-04-28T00:00:00.0000000+00:00) [FAIL]
      System.ComponentModel.Win32Exception : Error creating window handle.
      Stack Trace:
        /_/src/System.Windows.Forms/src/System/Windows/Forms/NativeWindow.cs(491,0): at System.Windows.Forms.NativeWindow.CreateHandle(CreateParams cp)
        ...
```

Here's one of examples: https://dev.azure.com/dnceng/public/_build/results?buildId=620251&view=results, build log: [System.Windows.Forms.Tests_netcoreapp5.0_x64 (20).log](https://github.com/dotnet/winforms/files/4542863/System.Windows.Forms.Tests_netcoreapp5.0_x64.20.log)
There is no code change, only an MD file. Debug build has succeeded, whilst Release has failed.

**Expected behavior:**

Tests pass without running out of handles.

**Minimal repro:**

N/A, builds fail intermittently.

**Comment by RussKie** [comment]
@JeremyKuhne any thoughts?

@MattGal is there a limit on handles we have on public agents?

**Comment by MattGal** [comment]
> @MattGal is there a limit on handles we have on public agents?

The actual limit seems to be hard to find documented, can you point me at that?

Either way, for our Windows machines in general we don't set any limits (this is a Windows limit) and you're running on a 4-core Windows-10 SKU machine with 16 GB of RAM so unlikely a resource constraint.  Further, machines are rebooted between every build leg to avoid problems like a test that leaks windows, handles, or processes that we fail to clean up.

I would expect this should be reproducible locally;  if you're stumped on this I can try to help out.

**Comment by MattGal** [comment]
@RussKie here's an idea.   See https://github.com/dotnet/winforms/pull/1025 .  You don't seem to be killing dhandler in your builds, and you're doing UI-based testing so it seems very relevant.  

Would also explain the lack of a local repro.

**Comment by MattGal** [comment]
Adding @sharwell who I think might be able to comment about window handle limits and this sort of automation.

**Comment by sharwell** [comment]
You should be able to run the tests locally with PerfView running, and get a summary of the handles which are released non-deterministically (e.g. via finalizers). Hopefully one type of handle stands out so it can be closed explicitly during the tests to avoid this situation.

**Comment by RussKie** [comment]
Thank you. I expect after #3064 is complete we should be leaking (too many) handles. Will probably add some sort of a trace in `~Control()` to see where else we may be leaking.

> You don't seem to be killing dhandler in your builds, and you're doing UI-based testing so it seems very relevant.
> 
> Would also explain the lack of a local repro.

Aye, you mentioned it before we migrated to the managed pool. I'll tweak the pipeline.

**Comment by RussKie** [comment]
I presume it is addressed by #3064.
